tests/ptf/fabric_test.py

```diff
@@ -1248,22 +1435,21 @@ def runDownstreamV4Test(self, pkt, in_tagged, line_enabled):
         self.setup_bng()
         self.setup_line_v4(
             s_tag=s_tag, c_tag=c_tag, line_id=line_id, ipv4_addr=pkt[IP].dst,
-            pppoe_session_id=pppoe_session_id, ds_next_id=next_id,
-            enabled=line_enabled)
+            mac_src=pkt[Ether].src, pppoe_session_id=pppoe_session_id, enabled=line_enabled)
 
         # Build expected packet from the input one, we expect it to be routed
         # and encapsulated in double VLAN tags and PPPoE.
-        exp_pkt_base = pkt_add_pppoe(pkt, type=1, code=PPPOE_CODE_SESSION_STAGE,
-                                     session_id=pppoe_session_id)
-        exp_pkt_base = pkt_add_vlan(exp_pkt_base, vlan_vid=vlan_id_inner)
-        exp_pkt_base = pkt_add_vlan(exp_pkt_base, vlan_vid=vlan_id_outer)
+        exp_pkt = pkt_add_pppoe(pkt, type=1, code=PPPOE_CODE_SESSION_STAGE, session_id=pppoe_session_id)
+        exp_pkt = pkt_add_vlan(exp_pkt, vlan_vid=vlan_id_inner)
+        exp_pkt = pkt_add_vlan(exp_pkt, vlan_vid=vlan_id_outer)
```

Comment:
it's unclear why you need the new function `pkt_add_innver_vlan` if you don't use it